src/Selenium2Library/keywords/_browsermanagement.py

```
@@ -27,6 +34,34 @@
                  'edge': "_make_edge"
                 }
 
+class _ReusableDriver(webdriver.Remote):
+    def __init__(self, command_executor, sid, keep_alive=False, file_detector=None):
+
+        self.reuse_session_id = sid
+        super(_ReusableDriver, self).__init__(command_executor, desired_capabilities={},
+                                              browser_profile=None,
+                                              proxy=None, keep_alive=keep_alive,
+                                              file_detector=file_detector)
+
+    def stop_client(self):
+        if self.pid is not None and self.pid != '':
+            logger.info("Stopping driver for session '%s' with PID: %s" % (self.session_id, self.pid))
+            try:
+                os.kill(int(self.pid), signal.SIGTERM)
+            except Exception:
```

1) If the pid does not exist then the different OS will raise different types of errors. In Windows 10 it is `WindowsError` and in Linux is it `OSError`. But both exceptions are based on the same error: `EnvironmentError`. Perhaps it would be best to do: `expect EnvironmentError`
2) If we can not kill the driver executable, is there any reason to continue? Should we just let the exception raise and just stop the test execution?

---

1 done
2 I'd say no. The side-effect is that the affected browser just stays open and is not anymore "reachable" via Selenium and must be closed manually. IMHO this is okay when considering the use cases for these KW's = TC development i.e. not for production use.